src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs

```diff
@@ -38,5 +88,59 @@ public string GetString(string resource)
 
 		// TODO: Remove this property when getting rid of ResourceHelper
 		public static Func<string, string> GetStringInternal { get; set; }
+
+		/// <summary>
+		/// Registers an assembly for resources lookup
+		/// </summary>
+		/// <param name="assembly">The assembly containing upri resources</param>
+		public static void AddLookupAssembly(Assembly assembly)
+		{
+			foreach (var name in assembly.GetManifestResourceNames())
+			{
+				if (name.EndsWith(".upri"))
+				{
+					ProcessResourceFile(name, assembly.GetManifestResourceStream(name));
+				}
+			}
+		}
+
+		public static string DefaultLanguage { get; set; }
+
+		internal static void ClearResources()
+		{
+			_resources.Clear();
+		}
+
+		internal static void ProcessResourceFile(string name, Stream input)
+		{
+			using (var reader = new BinaryReader(input))
+			{
+				if (!reader.ReadBytes(3).SequenceEqual(new byte[] { 0x75, 0x6E, 0x6F }))
```

Comment:
the same magic?

Reply:
That kind of header is often called a "magic header". So you're right.